src/lib/pk_pad/iso9796/iso9796.cpp

```diff
@@ -46,7 +46,7 @@ std::vector<uint8_t> iso9796_encoding(const std::vector<uint8_t>& msg,
       msg1 = std::vector<uint8_t>(msg.begin(), msg.begin() + capacity);
       hash->update(std::span(msg).subspan(capacity));
    } else {
-      msg1 = msg;
+      msg1.assign(msg.begin(), msg.end());
    }
```

Comment:
This could perhaps be re-phrased  now (GitHub doesn't let me comment the entire hunk):

```C++
const auto prefix = std::min(capacity, msg.size());
const auto msg1 = msg.first(prefix)
hash->update(msg.subspan(prefix)); // <= may be empty
const auto hmsg2 = hash->final_stdvec();
```

Disclaimer: I didn't test this.

Reply:
Did it in a slightly different way, but yeah there was no need for this to be copied to a vector